src/benchmarklib/benchmark_config.cpp

```diff
@@ -0,0 +1,53 @@
+#include "benchmark_config.hpp"
+
+namespace opossum {
+
+BenchmarkConfig::BenchmarkConfig(const BenchmarkMode benchmark_mode, const bool verbose, const ChunkOffset chunk_size,
+                                 const EncodingConfig& encoding_config, const size_t max_num_query_runs,
+                                 const Duration& max_duration, const Duration& warmup_duration, const UseMvcc use_mvcc,
+                                 const std::optional<std::string>& output_file_path, const bool enable_scheduler,
+                                 const uint32_t cores, const uint32_t clients, const bool enable_visualization,
+                                 const bool cache_binary_tables, std::ostream& out)
+    : benchmark_mode(benchmark_mode),
+      verbose(verbose),
+      chunk_size(chunk_size),
+      encoding_config(encoding_config),
+      max_num_query_runs(max_num_query_runs),
+      max_duration(max_duration),
+      warmup_duration(warmup_duration),
+      use_mvcc(use_mvcc),
+      output_file_path(output_file_path),
+      enable_scheduler(enable_scheduler),
+      cores(cores),
+      clients(clients),
+      enable_visualization(enable_visualization),
+      cache_binary_tables(cache_binary_tables),
```

Comment:
> Tables will be loaded from binary files whether the flag is set or not. I found this the most intuitive behaviour, please let me know if you disagree (e.g. @mrks)

I don't like this. If I load my tables from CSVs and they are cached as binaries, my changes to those CSVs would not be reflected. I can almost guarantee that people will lose chunks of hairs over this. Explicitly setting a flag would not immediately solve this either.

Maybe we could store the MD5 of the used CSV file. Is that an option?

Reply:
mmmh, changing the binary format is something that I don't like doing... :wink: 

Would updating the binary file if it is older than the csv do it for you?

Reply:
I wouldn't touch the binary format for that as there will be binaries that are not generated from a CSV. Maybe store it in the filename?